src/qibolab/instruments/zhinst/executor.py

```diff
@@ -0,0 +1,741 @@
+            modulation_type=lo.ModulationType.SOFTWARE,
+        )
+        threshold = None
+
+        if options.acquisition_type == AcquisitionType.DISCRIMINATION:
+            if qubit.kernel is not None:
+                # Kernels don't work with the software modulation on the acquire signal
+                oscillator = None
+            else:
+                # To keep compatibility with angle and threshold discrimination (Remove when possible)
+                threshold = qubit.threshold
+
+        self.calibration[f"/logical_signal_groups/q{q}/acquire_line"] = (
+            lo.SignalCalibration(
+                oscillator=oscillator,
+                range=qubit.feedback.power_range,
+                port_delay=self.time_of_flight * NANO_TO_SECONDS,
+                threshold=threshold,
+            )
+        )
+
+    def register_drive_line(self, qubit, intermediate_frequency):
+        """Registers qubit drive line to calibration and signal map."""
+        q = qubit.name  # pylint: disable=C0103
+        self.signal_map[qubit.drive.name] = self.device_setup.logical_signal_groups[
+            f"q{q}"
+        ].logical_signals["drive_line"]
+        self.calibration[f"/logical_signal_groups/q{q}/drive_line"] = (
+            lo.SignalCalibration(
+                oscillator=lo.Oscillator(
+                    frequency=intermediate_frequency,
+                    modulation_type=lo.ModulationType.HARDWARE,
+                ),
+                local_oscillator=lo.Oscillator(
+                    frequency=int(qubit.drive.local_oscillator.frequency),
+                ),
+                range=qubit.drive.power_range,
+                port_delay=None,
+                delay_signal=0,
+            )
+        )
+
+    def register_flux_line(self, qubit):
+        """Registers qubit flux line to calibration and signal map."""
+        q = qubit.name  # pylint: disable=C0103
+        self.signal_map[qubit.flux.name] = self.device_setup.logical_signal_groups[
+            f"q{q}"
+        ].logical_signals["flux_line"]
+        self.calibration[f"/logical_signal_groups/q{q}/flux_line"] = (
+            lo.SignalCalibration(
+                range=qubit.flux.power_range,
+                port_delay=None,
+                delay_signal=0,
+                voltage_offset=qubit.flux.offset,
+            )
+        )
+
+    def register_couplerflux_line(self, coupler):
+        """Registers qubit flux line to calibration and signal map."""
+        c = coupler.name  # pylint: disable=C0103
+        self.signal_map[coupler.flux.name] = self.device_setup.logical_signal_groups[
+            f"qc{c}"
+        ].logical_signals["flux_line"]
+        self.calibration[f"/logical_signal_groups/qc{c}/flux_line"] = (
+            lo.SignalCalibration(
+                range=coupler.flux.power_range,
+                port_delay=None,
+                delay_signal=0,
+                voltage_offset=coupler.flux.offset,
+            )
+        )
```

Comment:
Could they be merged into a single function?

Reply:
You would need still need to check if it is a coupler or qubit inside the function:

```python
    def register_flux_line(self, qubit):
        """Registers flux line to calibration and signal map."""
        q = qubit.name  # pylint: disable=C0103
        name = "q" if isinstance(qubit, Qubit) else "qc"
        self.signal_map[qubit.flux.name] = self.device_setup.logical_signal_groups[
            f"{name}{q}"
        ].logical_signals["flux_line"]
        self.calibration[f"/logical_signal_groups/{name}{q}/flux_line"] = (
            lo.SignalCalibration(
                range=qubit.flux.power_range,
                port_delay=None,
                delay_signal=0,
                voltage_offset=qubit.flux.offset,
            )
        )
```

and maybe change qubit to `quantum_element` or whathever so it looks less confusing ?

Reply:
they could be combined, but these functions are going to change anyways, because of 0.2, so I decided not to touch them for now